doc/src/conf.py

```diff
@@ -363,26 +363,20 @@
 
 # Additional stuff for the LaTeX preamble.
 # Tweaked to work with XeTeX.
+latex_engine = 'xelatex'
 latex_elements = {
-    'babel':     '',
-    'fontenc': r'''
-% Define version of \LaTeX that is usable in math mode
-\let\OldLaTeX\LaTeX
-\renewcommand{\LaTeX}{\text{\OldLaTeX}}
-
+    'babel': r'\usepackage[english]{babel}',
+    'fontpkg': r'''
 \usepackage{bm}
-\usepackage{amssymb}
-\usepackage{fontspec}
-\usepackage[english]{babel}
 \defaultfontfeatures{Mapping=tex-text}
 \setmainfont{DejaVu Serif}
 \setsansfont{DejaVu Sans}
 \setmonofont{DejaVu Sans Mono}
 ''',
-    'fontpkg':   '',
-    'inputenc':  '',
-    'utf8extra': '',
     'preamble':  r'''
+% Define version of \LaTeX that is usable in math mode
+\let\OldLaTeX\LaTeX
```

Comment:
```suggestion
\usepackage{letltxmacro}
\LetLtxMacro\OldLaTeX\LaTeX
```

Reply:
Hmm, no. I will revert that. I did not want to spend so much time into debugging this alas I did.

First two side remarks:

1. The same issue I have at my locale with chromium are [seen](https://github.com/sympy/sympy/actions/runs/10167896565/job/28122085075?pr=26868#step:6:1574)
2. With current LaTeX there is no build crash with the OldLaTeX thing whatever its position in the preamble.

The test-sphinx here is done with TeXLive 2019, LaTeX of 2020-02-02.  It turns out that what matters is if the redefinition is done before or after `\usepackage[english]{babel}` is encountered.

The culprit is babel who did `\AtBeginDocument{\DeclareTextCommandDefault {\LaTeX }{\org@LaTeX }}` after having stored the meaning of `\LaTeX` into `\org@LaTeX`. When I moved the (doubtful because `\LaTeX` is a robust macro and was not handled as such) hack with  `\OldLaTeX` later in preamble, this caused `\org@LaTeX` to have preserved the old meaning which is the robust one: `\protect \LaTeX<space in the name>`. When the `\OldLaTeX` is earlier in preamble, the meaning of `\org@LaTeX` is `\text {\OldLaTeX }`. (which is what we want).

When I tried the `\LetLtxMacro` thinking there was some problem with robust macros and not wanting to investigate more, I thought issue was solved because it builds at my place and I never had made a complete build before (and I thought the problem came from toc file).  But the point is that with current babel and LaTeX and hyperref, things are different, the meaning is not modified by babel. Back with TeXLive 2019 as used for the GH action, the business with `\LetLtxMacro\OldLaTeX\LaTeX` served nothing anyhow, because `\LaTeX` was still redefined at begin document by babel to be `\protect\LaTeX<space in the name>` and `\LaTeX<space in the name>` was not modified by my code at all.  In current LaTeX, the action of `babel` has changed it does now `\DeclareTextCommandDefault {\LaTeX }{\ensureascii {\org@LaTeX }}` but it is nightmare to trace the LaTeX code at `\begin{document}` which occupies now tens of thousands of lines. Also `hyperref` has changed.  So it is very complicated to understand why with current LaTeX after `\begin{document}` we have `\LaTeX` expanding to `\text{\OldLaTeX}` as was expected.

Anyway in order to solve this on old LaTeX, we must make sure not only not to use `\let` (to avoid potential problems if `\LaTeX` is in a title ending up in `.aux` then `.toc` file) but also to delay at `\begin{document}` the redefinition of `\LaTeX` to override the fact that `babel` in those days redefined it there.

I will do that next.